profiler/options.go

```diff
@@ -197,15 +198,28 @@ func WithAgentAddr(hostport string) Option {
 	}
 }
 
-// WithAPIKey is deprecated and might be removed in future versions of this
-// package. It allows to skip the agent and talk to the Datadog API directly
-// using the provided API key.
+// WithAPIKey is deprecated and will be ignored unless you also use the
+// WithDeprecatedAPIKey() option which has more information on the deprecation.
 func WithAPIKey(key string) Option {
 	return func(cfg *config) {
 		cfg.apiKey = key
 	}
 }
 
+// WithDeprecatedAPIKey exists to enable the deprecated agentless upload mode.
```

Comment:
We could also call this `WithDeprecatedAgentlessMode` or something. I just think `WithAgentlessUpload` doesn't make this option sound scary enough to raise red flags with customers.

Reply:
Why would you add a new option which is deprecated straight out of the box? I'm pretty sure you guys talked about this forehand and I'm out of context, but just putting in an outsider perspective.